### PR TITLE
libxx: Update README

### PR DESCRIPTION
**Pull request by yamt**
## Summary
This doc seems a bit outdated to me.
This commit is my best attempt to update it.

### CODE REVIEW

File: libs/libxx/README.txt [hunk 1]
```diff
@@ -1,28 +1,44 @@
...
-
- - void *operator new(std::size_t nbytes);
- - void operator delete(void* ptr);
- - void operator delete[](void *ptr);
- - void __cxa_pure_virtual(void);
- - int __aeabi_atexit(void* object, void (*destroyer)(void*), void *dso_handle);
- - int __cxa_atexit(__cxa_exitfunc_t func, FAR void *arg, FAR void *dso_handle);
+There is another version of uClibc++ that has been customized for
+use with NuttX. That version that can be found in the NuttX
+uClibc++ GIT repository.  (https://bitbucket.org/nuttx/uclibc/src/master/)
```
Comment (xiaoxiang781216) [inline libs/libxx/README.txt:39]: with the latest nuttx enhancement, the customized uClibc++ doesn't need anymore, should we remove this version from README?

File: libs/libxx/README.txt [hunk 2]
```diff
@@ -1,28 +1,44 @@
...
+ - uClibc++ C++ library (http://cxx.uclibc.org/)
+   This implementation is selected with CONFIG_UCLIBCXX=y.
+
+If you have a need for more extensive C++ support, the following
+libraries might be usable too:
 
  - libstdc++  (part of GCC)
  - STLport    http://www.stlport.org/
- - uClibc++   http://cxx.uclibc.org/
  - uSTL       http://ustl.sourceforge.net/
```
Comment (xiaoxiang781216) [inline libs/libxx/README.txt:35]: Should we remove STLPort and uSTL? these two project stop maintain if I remeber correctly.

File: libs/libxx/README.txt (hunk 1 above)
Comment (yamt) [reply libs/libxx/README.txt:39]: maybe. i don't know the state of the repo.

File: libs/libxx/README.txt (hunk 2 above)
Comment (yamt) [reply libs/libxx/README.txt:35]: i have no idea.

Comment (v01d) [reply libs/libxx/README.txt:35]: I would remove all of those, as we're not really officially supporting them. The three options right now I think that are really sufficient.

Comment (yamt) [reply libs/libxx/README.txt]: done

File: libs/libxx/README.txt (hunk 1 above)
Comment (yamt) [reply libs/libxx/README.txt]: done

**Comment by btashton** [comment]
@yamt would you want to fix the sim readme for cxxtest as well that was part of #2086  I think there is also old/wrong information there about the state libxx.  If you want to defer that as well that is fine, I just really did not feel comfortable making the updates.

**Comment by yamt** [comment]
@btashton i haven't looked at uClibc++ yet. only other twos.

**Comment by btashton** [comment]
Thanks for all the work on the C++ support, I'll bring this in now.  We should probably add some real docs / a guide to the published documentation about the recommended way to use C++ at some point.

File: libs/libxx/README.txt (hunk 1 above)
Comment (btashton) [reply libs/libxx/README.txt]: @xiaoxiang781216 this same out dated information in in the SIM readme as well...

**Pull request merged**